Add accessibility statement (#3)
Quansight-Labs/accessibility: statement.md

# JupyterLab Accessibility Statement

## Edited from the [W3C accessibility statement generator](https://www.w3.org/WAI/planning/statements/generator/#create)

> ## ⚠️ Draft Accessibility Statement for JupyterLab

Reviewer: ```suggestion
## :construction: Draft Accessibility Statement for JupyterLab
```
I have realised that the other emoji renders widely different across OS and editors and is a nightmare

Author: I also know the `:emoji:` format doesn't render everywhere, but we can do this if it's preferred.

> This is an accessibility statement from <span class="basic-information organization-name">Jupyter accessibility contributors</span>.

Reviewer: There seems to be a few `<span>` and the such straggling - do they need removing?

### Measures to support accessibility

Jupyter accessibility contributors take the following measures to ensure accessibility of JupyterLab:

* Include accessibility as part of our mission statement.
* Provide continual accessibility training for our community.
* Assign clear accessibility goals and responsibilities.

> * Employ formal accessibility quality assurance methods.

Reviewer: Idea: we should add something on documenting - changes, improvements and approaches to testing and QA

Author: These were from a list of W3C suggested ones, so I'm unclear if they map to some kind of special requirements.

I added another point.

### Conformance status

> The <a href="https://www.w3.org/WAI/standards-guidelines/wcag/">Web Content Accessibility Guidelines (WCAG)</a> defines requirements for designers and developers to improve accessibility for people with disabilities. It defines three levels of conformance: Level A, Level AA, and Level AAA. JupyterLab is non conformant with WCAG 2.1 level AA. Non conformant means that the content does not conform the accessibility standard.

Reviewer: Is this the auto-generated text? 

> Non conformant means that the content does not conform the
Using the word in the definition which makes it hard to follow/understand

Reviewer: Stephanie Walter has an amazing illustration for this in https://stephaniewalter.design/blog/5-illustrations-to-understand-and-promote-accessibility/

Could we add this to add some visual sugar - of course with due acknowledgement and x-linking to the post

Author: It is the auto-generated text; most of this is not my work alone.

I'll look into how I need to add the image with credit and add it.

### Feedback

> We welcome your feedback on the accessibility of JupyterLab. Please let us know if you encounter accessibility barriers on JupyterLab:

Reviewer: accessibility status? approach?

* [Write an issue on jupyter/accessibility](https://github.com/jupyter/accessibility/issues/new)
* [Write an issue on jupyterlab/jupyterlab](https://github.com/jupyterlab/jupyterlab/issues/new) and request it be labeled [tag:accessibility](https://github.com/jupyterlab/jupyterlab/labels/tag%3AAccessibility)

### Compatibility with browsers and assistive technology

JupyterLab is designed to be compatible with the following technologies:

> * Windows, macOS, iOS, Android; Firefox, Chrome, Safari, Chromium browsers (mobile and desktop).

Reviewer 2: Oh wow, I've never opened JupyterLab on my smartphone. I assumed it wouldn't work...

JupyterLab is not compatible with:

* Internet Explorer, Edge; JAWS, NVDA, VoiceOver, Narrator, Orca screen readers; voice control technology.

### Technical specifications

Accessibility of JupyterLab relies on the following technologies to work with the particular combination of web browser and any assistive technologies or plugins installed on your computer:

* HTML
* WAI-ARIA
* CSS
* JavaScript

These technologies are relied upon for conformance with the accessibility standards used.

### Limitations and alternatives

Despite our best efforts to ensure accessibility of JupyterLab, there may be some limitations. Below is a description of known limitations, and potential solutions. Please contact us if you observe an issue not listed below.

> Known limitations for JupyterLab:

Reviewer: ```suggestion
**Known limitations for JupyterLab:**
```

> 1. **Documents**: Documents writen by the community may not include accessible content because we do not and cannot review every document that can be opened in JupyterLab. To support accessible documents, we are drafting guidelines for accessible document content with an emphasis on Jupyter notebooks. Please report the issue to the author and [open an issue on jupyter/accessibility](https://github.com/jupyter/accessibility/issues/new) describing the problem and the behavior you expect so we may integrate it into our content guidelines.

Reviewer: Maybe add an example: "Documents such as notebooks, markdown files, etc."

> 1. **Documents**: Documents writen by the community may not include accessible content because we do not and cannot review every document that can be opened in JupyterLab. To support accessible documents, we are drafting guidelines for accessible document content with an emphasis on Jupyter notebooks. Please report the issue to the author and [open an issue on jupyter/accessibility](https://github.com/jupyter/accessibility/issues/new) describing the problem and the behavior you expect so we may integrate it into our content guidelines.

Reviewer: ```suggestion
1. **Documents**: Documents written by the community may not include accessible content because we do not and cannot review every document that can be opened and edited in JupyterLab. To support accessible documents, we are drafting guidelines for accessible document content with an emphasis on Jupyter notebooks. Please report the issue to the author and [open an issue on jupyter/accessibility](https://github.com/jupyter/accessibility/issues/new) describing the problem and the behavior you expect so we may integrate it into our content guidelines.
```

> 2. **JupyterLab extensions**: JupyterLab extensions writen by the community may not be accessible because JupyterLab extensions can be written by anyone in the community and have no required review process. We do not and can not review every JupyterLab extension. To support accessible extensions, we encourage extension authors to use existing, accessible JupyterLab components for their extension. We also  provide periodic opportunities for community education on accessibility. Please report the issue to the author and let them know the [jupyter/accessibility](https://github.com/jupyter/accessibility/) community may be able to provide guidance.

Reviewer: ```suggestion
2. **JupyterLab extensions**: JupyterLab extensions written by the community may not be accessible because JupyterLab extensions can be written by anyone in the community and have no standard review process. We do not and can not review every JupyterLab extension. To support accessible extensions, we encourage extension authors to use existing, accessible JupyterLab components for their extensions. We also provide periodic opportunities for community education on accessibility. Please report the issue to the author and let them know the [jupyter/accessibility](https://github.com/jupyter/accessibility/) community may be able to provide guidance.
```

### Assessment approach

Jupyter accessibility contributors assessed the accessibility of JupyterLab by the following approaches:

* Self-evaluation
* Regular automated testing to monitor for regressions (can be found at [jupyter/accessibility](https://github.com/jupyter/accessibility)

> * User feedback</li>

Reviewer: ```suggestion
* User feedback
```
assuming that was left over

### Evaluation report

An evaluation for JupyterLab is available at: [jupyterlab/jupyterlab/issues/9399](https://github.com/jupyterlab/jupyterlab/issues/9399).

> ### Evaluation report

Reviewer 2: Two headings with same title?

Author: Yep, that's what the generator made. 🤷

> ### Evaluation report

Reviewer: ```suggestion
```

User reports on JupyterLab's accessibility are available at:[the jupyterlab/jupyterlab label `tag:accessibility`](https://github.com/jupyterlab/jupyterlab/labels/tag%3AAccessibility).

> ### Formal complaints

Reviewer 2: Merge with Feedback? Feedback and Formal Complaints?

Author: I think they are separate because formal complaints seems to be private and/or legal action related complaints, but we don't have a more formal channel at the moment so I'll put them together.

> * [Write an issue on jupyter/accessibility](https://github.com/jupyter/accessibility/issues/new)

Reviewer: Shall we create a template for this? or just rely on a blank issue?

Author: I think adding a template could be useful, but I don't have one to add at the moment. Do you want me to hold off on this PR until that's done?

There's a list of templates that I believe are wanted for that repo, so I would probably do a few at once.

Reviewer: Let's move templates to the next cycle and get this merged

> * [Write an issue on jupyterlab/jupyterlab](https://github.com/jupyterlab/jupyterlab/labels/tag%3AAccessibility) and request it be labeled tag:accessibility.

Reviewer 2: As an end user, how should I know which repo to submit my issue to?

Author: I don't know, because I don't know what the difference would be in opening it on one repo or the other.

My estimate is that

Writing on jupyterlab/jupyterlab
- Would likely be the place where the change would need to be made (or in a related repo like Lumino)
- Would be the place where we may be able to get support from people not as focused on accessibility but knowledgeable about JupyterLab
- Has lots and lots of issues and could risk burying new issues in short amounts of time
- To be honest, I don't follow all JupyterLab issues by default because there's too much activity

Writing on jupyter/accessibility
- Would give a lot of attention to an issue
- Would likely require opening another issue on the repo where the code changes need to be made (at some point)
- Has less people actively following and may not have the fastest support

Does this make you feel more confident if we should advise only one? Or how we should tell people to choose?

Reviewer 2: So I do think we should advise only one, to keep things simple. But I too am not entirely sure which one we should use.

I think for me, a big part of it is about where the accessibility statement lives. If it's in the jupyterlab repo, then I think we should point them to create issues on the jupyterlab repo (with accessibility tag, just like you're doing). If the statement lives in the accessibility repo, then they should create issues on the accessibility repo. I dunno. I don't feel too strongly about that. 

But I do think it's pretty confusing to give somebody two options without even a hint about why they should choose one over the other. 

I think our own lack of razor sharp clarity on this bleeds through to the accessibility statement. What I mean is, we don't seem to have ourselves a clear criteria for when to submit to one or the other. A clear criteria, for example, would be something like: for all accessibility issues pertaining to JupyterLab create an issue on the JupyterLab repo; for all other Jupyter projects (JupyterHub, Lumino, etc.) submit an issue to the accessibility repo.

Reviewer: I thought we had decided to keep the statement in this repo and x-reference to it from the Jupyterlab readme/docs?

Suggestion - even if these sections have been merged

```
* For accessibility-related matters within the broader Jupyter ecosystem or directly related with this statement - [Write an issue on jupyter/accessibility](https://github.com/jupyter/accessibility/issues/new)
* For accessibility issues related to JupyterLab only -[Write an issue on jupyterlab/jupyterlab](https://github.com/jupyterlab/jupyterlab/labels/tag%3AAccessibility) and request it be labeled tag:accessibility.
```

# JupyterLab Accessibility Statement

## Edited from the [W3C accessibility statement generator](https://www.w3.org/WAI/planning/statements/generator/#create)

> ## :construction: Draft Accessibility Statement for JupyterLab

Reviewer: Thanks for the reminder here @isabela-pf - I always forget there are a million md subsets - replacing with the emoji
```suggestion
## 🚧 Draft Accessibility Statement for JupyterLab
```

### Assessment approach

Jupyter accessibility contributors assessed the accessibility of JupyterLab by the following approaches:

* Self-evaluation
* Regular automated testing to monitor for regressions (can be found at [jupyter/accessibility](https://github.com/jupyter/accessibility)
* User feedback

### Evaluation report

An evaluation for JupyterLab is available at: [jupyterlab/jupyterlab/issues/9399](https://github.com/jupyterlab/jupyterlab/issues/9399).

User reports on JupyterLab's accessibility are available at:[the jupyterlab/jupyterlab label `tag:accessibility`](https://github.com/jupyterlab/jupyterlab/labels/tag%3AAccessibility).

### Date

> This statement was created on 16 May 2022 using the [W3C Accessibility Statement Generator Tool](https://www.w3.org/WAI/planning/statements/) with additions and edits from the Jupyter accessibility contributors community.

Reviewer: ```suggestion
This statement was created on 16 May 2022 using the [W3C Accessibility Statement Generator Tool](https://www.w3.org/WAI/planning/statements/) with additions and edits from the Jupyter accessibility contributors community.

```
end of file line

*By [Stéphanie Walter](https://stephaniewalter.design) ([Source](https://stephaniewalter.design/blog/5-illustrations-to-understand-and-promote-accessibility/))*

JupyterLab's accessibility does not exist in isolation. JupyterLab inherits much from the many projects it is built upon and its accessibility conformance may impact projects built off of JupyterLab or its components. The accessibility of this ecosystem is interlinked, so conformance may need to be resolved at different levels in order to impact JupyterLab positively.

### Feedback and Formal complaints

We welcome your feedback and formal complaints on the accessibility status of JupyterLab. Please let us know if you encounter accessibility barriers on JupyterLab:

* [Write an issue on jupyter/accessibility](https://github.com/jupyter/accessibility/issues/new)
* [Write an issue on jupyterlab/jupyterlab](https://github.com/jupyterlab/jupyterlab/issues/new) and request it be labeled [tag:accessibility](https://github.com/jupyterlab/jupyterlab/labels/tag%3AAccessibility)

> * If you are interested in being part of any potential research or organized feedback initiatives, please ___. Gauging community interest in user research participation helps us gain the support to make it happen.

Author: @trallard mentioned pointing people to the repo, but I think we need something a little more specific. I'm also unsure if a totally public place is the best since we may want to collect contact info. This is up for discussion.

I'm also happy to remove this section and more forward without it if it becomes too messy of a discussion.

```suggestion
* If you are interested in being part of any potential research or organized feedback initiatives, please [comment on this issue](). Gauging community interest in user research participation helps us gain the support to make it happen.
```
(Also oops to the accidental formatting I triggered. I'll fix that when we agree on what to do next.)

Reviewer: I completely forgot - yes for now we can use your email and we should be able to request a Jupyter one too 🤔
